framework/meta/src/folder_structure/relevant_directory.rs

```diff
@@ -83,26 +83,26 @@ impl RelevantDirectories {
             .filter(|dir| dir.dir_type == DirectoryType::Contract)
     }
 
-    pub fn count_for_version(&self, version: &str) -> usize {
+    pub fn count_for_version(&self, version: &FrameworkVersion) -> usize {
         self.0
             .iter()
-            .filter(|dir| dir.version.semver == version)
+            .filter(|dir| dir.version.semver == *version)
             .count()
     }
 
     pub fn iter_version(
         &mut self,
-        version: &'static str,
+        version: &'static FrameworkVersion,
     ) -> impl Iterator<Item = &RelevantDirectory> {
         self.0
             .iter()
-            .filter(move |dir| dir.version.semver == version)
+            .filter(move |dir| dir.version.semver == *version)
     }
 
     /// Marks all appropriate directories as ready for upgrade.
-    pub fn start_upgrade(&mut self, from_version: &'static str, to_version: &'static str) {
+    pub fn start_upgrade(&mut self, from_version: &'static FrameworkVersion, to_version: &'static FrameworkVersion) {
```

Comment:
'static lifetime also not required here, I think

Reply:
Can't remove 'static lifetime due to error "does not live long enough"